checker/checker.go

```diff
@@ -144,6 +146,27 @@ func (v *visitor) BinaryNode(node *ast.BinaryNode) reflect.Type {
 	l := v.visit(node.Left)
 	r := v.visit(node.Right)
 
+	// check operator overloading
+	if fns, ok := v.operators[node.Operator]; ok {
+		for _, fn := range fns {
+			fnType, ok := v.types[fn]
+			if !ok || fnType.Type.Kind() != reflect.Func {
+				continue
+			}
+
+			if fnType.Type.NumIn() != 2 || fnType.Type.NumOut() != 1 {
```

Comment:
Here is again, we can check this before checker phase, in "configuration" phase.
And no need to wait on then a proper expression will be compiled to get an error.